architectures/firmware/sof-zephyr/rtos_layer/io_drivers/index.rst

```diff
@@ -0,0 +1,17 @@
+.. _io_drivers:
+
+IO Drivers
+##########
+
+The IO Drivers provide access to an IO HW Interfaces connected to the DSP, e.g.
+I2S, DMIC, etc. and are managed as a part of Zephyr RTOS. The Audio IO Drivers
+share generic `Zephyr DAI interface <https://docs.zephyrproject.org/apidoc/latest/group__dai__interface.html>`__.
+For a full list of IO drivers available on the specific platform, refer to
+:ref:`platforms`. HW IO is accessed via the `Gateway` interface inside the FW.
+The actual implementation of that interface depends on the underlying HW IO
+mechanism. Gateways uses Zephyr DMA interface to transmit the data to/from
```

Comment:
little bit unsure about the gateway concept... :) I mean in the code it is still a "dai" counterpart struct to zephyr dai..

Reply:
The gateways code development is in progress, so we can wait with this PR till code is complete. I wanted to share architecture description so it can give a better understanding on the changes that are being introduced to the code.